src/Nethermind/Nethermind.DataMarketplace.Consumers/Shared/Services/ConsumerServicesBackgroundProcessor.cs

```diff
@@ -200,15 +208,15 @@ private async Task TryConfirmDepositsAsync(IReadOnlyList<DepositDetails> deposit
             }
         }
 
-        private async Task TryClaimRefundsAsync(IReadOnlyList<DepositDetails> deposits)
+        private async Task TryClaimRefundsAsync(IEnumerable<DepositDetails> deposits)
         {
```

Comment:
changing it from list to enumerable causes multiple enumerations, might be very expensive here if the reading is connected with db access